src/convenience/frameworks.node.ts

```diff
@@ -1,90 +1,85 @@
 // deno-lint-ignore-file no-explicit-any
-import type { IncomingMessage, ServerResponse } from "http";
+import { IncomingMessage, ServerResponse } from "http";
+import safeStrCmp from "tsscmp";
+import { AdapterFactory } from "./webhook";
 
 /**
  * HTTP Web frameworks for which grammY provides compatible callback out of the
  * box.
  */
-export type SupportedFrameworks =
-    | "express"
-    | "http"
-    | "https"
-    | "koa"
-    | "fastify"
-    | "worktop"
-    | "aws-lambda";
 
-export const defaultFramework: SupportedFrameworks = "express";
-
-export const frameworkAdapters = {
-    express: (req: any, res: any) => ({
-        update: Promise.resolve(req.body),
-        end: () => res.end(),
-        respond: (json: string) => {
-            res.set("Content-Type", "application/json");
-            res.send(json);
-        },
-    }),
-    http: (req: IncomingMessage, res: ServerResponse) => ({
-        update: new Promise<any>((resolve) => {
+export const http: AdapterFactory<{ secretPath: string }> = ({ secretPath }) =>
+    (req: IncomingMessage, res: ServerResponse) => ({
+        update: new Promise((resolve) => {
+            if (
+                req.method !== "POST" || !safeStrCmp(secretPath, req.url ?? "")
+            ) {
+                resolve(undefined);
+            }
             const chunks: Buffer[] = [];
             req
-                .on("data", (chunk: Buffer) => chunks.push(chunk))
+                .on("data", (chunk) => chunks.push(chunk))
                 .on("end", () => {
                     const raw = Buffer.concat(chunks).toString("utf-8");
                     resolve(JSON.parse(raw));
                 });
         }),
         end: () => res.end(),
-        respond: (json: string) => {
-            return res
+        respond: (json) =>
+            res
                 .writeHead(200, { "Content-Type": "application/json" })
-                .end(json);
-        },
-    }),
-    https: (req: IncomingMessage, res: ServerResponse) => ({
-        update: new Promise<any>((resolve) => {
-            const chunks: Buffer[] = [];
-            req
-                .on("data", (chunk: Buffer) => chunks.push(chunk))
-                .on("end", () => {
-                    const raw = Buffer.concat(chunks).toString("utf-8");
-                    resolve(JSON.parse(raw));
-                });
-        }),
+                .end(json),
+    });
+
+export const https = http;
+
+export const express: AdapterFactory = () =>
+    (req: any, res: any) => ({
```

Comment:
I'm wondering now if we should start importing the types of express as a development dependency in order to have type safety here … what's your opinion?

Reply:
Then we need to have `@types/*` for every framework used there, which is a bit overkill for a few lines of bridging code imo. If you're ok with that, I'll add that too.

Reply:
That's true, I agree that it's overkill. I'm fine with not having type safety here.